extensions/inline-data-transfer/inline-data-transfer-core/src/main/java/org/eclipse/dataspaceconnector/transfer/inline/core/DataOperatorRegistryImpl.java

```diff
@@ -0,0 +1,48 @@
+package org.eclipse.dataspaceconnector.transfer.inline.core;
+
+import org.eclipse.dataspaceconnector.spi.types.domain.transfer.DataRequest;
+import org.eclipse.dataspaceconnector.transfer.inline.spi.DataOperatorRegistry;
+import org.eclipse.dataspaceconnector.transfer.inline.spi.DataReader;
+import org.eclipse.dataspaceconnector.transfer.inline.spi.DataStreamPublisher;
+import org.eclipse.dataspaceconnector.transfer.inline.spi.DataWriter;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+public class DataOperatorRegistryImpl implements DataOperatorRegistry {
```

Comment:
This could be registered by default in the CoreTransferExtension as it does not have any dependencies. This registry is an essential part and should always be there, even if the `InlineDataFLowController` is not used.